javacomparator/DiffGenerator.java

```diff
@@ -0,0 +1,457 @@
+        generateGlobalDiffReport(allDiffs);
+
+        ArrayList<RunReportData> allPages = new ArrayList<>();
+        allPages.add(generalReport);
+        allPages.add(docxReport);
+        allPages.add(pptxReport);
+        allPages.add(xlsxReport);
+        createRunSummary(allPages);
+    }
+
+    private void createXLSXHeading(Workbook workbook, Sheet sheet, String cellContent){
+        Font titleFont = workbook.createFont();
+        titleFont.setBold(true);
+        titleFont.setFontHeightInPoints((short) 20);
+        titleFont.setColor(IndexedColors.BLUE.getIndex());
+
+        CellStyle headerCellStyle = workbook.createCellStyle();
+        headerCellStyle.setAlignment(HorizontalAlignment.CENTER);
+
+        Row headerRow = sheet.getRow(0);
+        Cell cell = headerRow.createCell(0);
+        cell.setCellValue(cellContent);
+        headerCellStyle.setFont(titleFont);
+        cell.setCellStyle(headerCellStyle);
+    }
+
+    private void createXLSXSection(Workbook workbook,Sheet sheet,int rowNum,int colNum,String cellContent){
+        Font sectionFont = workbook.createFont();
+        sectionFont.setBold(true);
+        sectionFont.setFontHeightInPoints((short) 16);
+        sectionFont.setColor(IndexedColors.RED.getIndex());
+
+        CellStyle headerCellStyle = workbook.createCellStyle();
+        headerCellStyle.setAlignment(HorizontalAlignment.CENTER);
+
+        Row headerRow = sheet.getRow(rowNum);
+        Cell cell = headerRow.createCell(colNum);
+        cell.setCellValue(cellContent);
+        headerCellStyle.setFont(sectionFont);
+        cell.setCellStyle(headerCellStyle);
+    }
+
+    private void createXLSXSubSection(Workbook workbook, Sheet sheet, int rowNum, String cellContent) {
+        Font subSectionFont = workbook.createFont();
+        subSectionFont.setBold(true);
+        subSectionFont.setFontHeightInPoints((short) 12);
+        subSectionFont.setColor(IndexedColors.GREEN.getIndex());
+
+        CellStyle headerCellStyle = workbook.createCellStyle();
+
+        Row headerRow = sheet.getRow(rowNum);
+        Cell cell = headerRow.createCell(0);
+        cell.setCellValue(cellContent);
+        headerCellStyle.setFont(subSectionFont);
+        cell.setCellStyle(headerCellStyle);
+    }
+
+    private void createXLSXDataSection(Workbook workbook,Sheet sheet,int rowNum,int colNum,String cellContent) {
+        Font subSectionFont = workbook.createFont();
+        subSectionFont.setBold(true);
+        subSectionFont.setFontHeightInPoints((short) 12);
+        subSectionFont.setColor(IndexedColors.BLACK.getIndex());
+
+        CellStyle headerCellStyle = workbook.createCellStyle();
+
+        Row row = sheet.getRow(rowNum);
+        Cell cell = row.createCell(colNum);
+        cell.setCellValue(cellContent);
+        headerCellStyle.setFont(subSectionFont);
+        cell.setCellStyle(headerCellStyle);
+    }
+
+    private void createRunSummary(ArrayList<RunReportData> pages) {
```

Comment:
This function also seems too huge, please split this up into multiple functions to have better readability.